feat(community): new community page design

LeoMcA opened the pull request:
## Summary

<!--
  Please reference the issue you're solving here.
  Example: Fixes #1234.
-->

https://mozilla-hub.atlassian.net/browse/MP-1369

- Implements the new design, and uses a markdown file to power the content
- Markdown -> JSON build step not implemented, done by copying the markdown file into curriculum, building that, then copying the json back out and manually fixing it up
- We'll implement this build step in rari separately 

---

## How did you test this change?

- `yarn dev` and `yarn render:html`
- tested across :3000 and :5042
- tested across breakpoints
- tested across light/dark mode

caugner committed:
chore(community): empty page

caugner committed:
wip

caugner committed:
mobile design

caugner committed:
chore(community): add hero bg

caugner committed:
chore: progress

caugner committed:
chore(community): add Contributors circles

caugner committed:
chore(community): wip

LeoMcA committed:
Merge remote-tracking branch 'origin/main' into MP-1369-new-community-page

LeoMcA committed:
feat(community): implement simplified design

LeoMcA committed:
fix(community): ensure csp doesn't block circle background

fiji-flo committed:
fix(search): don't index non doc pages

LeoMcA committed:
Merge remote-tracking branch 'origin/main' into MP-1369-new-community-page

LeoMcA committed:
fix(community): design tweaks from design review

https://mozilla-hub.atlassian.net/browse/MP-1457

LeoMcA committed:
fix(community): stylelint

https://mozilla-hub.atlassian.net/browse/MP-1457

LeoMcA committed:
fix(community): fix links, update contributors in spotlight

https://mozilla-hub.atlassian.net/browse/MP-1458

https://mozilla-hub.atlassian.net/browse/MP-1446

https://mozilla-hub.atlassian.net/browse/MP-1447

LeoMcA committed:
fix(community): add gap between contributor sections on mobile

https://mozilla-hub.atlassian.net/browse/MP-1449

LeoMcA committed:
chore(community): revert webpack changes

LeoMcA committed:
Merge remote-tracking branch 'origin/main' into MP-1369-new-community-page

caugner commented on deployer/src/deployer/search/__init__.py:
```diff
@@ -217,8 +217,8 @@ def to_search(file, _index=None):
 
-    if doc["mdn_url"].startswith("/en-US/curriculum/"):
-        # Skip curriculum content for now.
+    if "/docs/" not in doc["mdn_url"]:
```
Does this really work?

LeoMcA committed:
fix(community): use avatars.githubusercontent.com directly to avoid cookie warnings

also add readme for build hack

LeoMcA committed:
fix(community): react item key error

LeoMcA committed:
Merge remote-tracking branch 'origin/main' into MP-1369-new-community-page

LeoMcA committed:
fix(community): match button padding to login button

https://mozilla-hub.atlassian.net/browse/MP-1457?focusedCommentId=932467

LeoMcA committed:
fix(community): cache github issues response in localstorage to mitigate ratelimiting

https://mozilla-hub.atlassian.net/browse/MP-1452

LeoMcA committed:
Merge remote-tracking branch 'origin/main' into MP-1369-new-community-page

LeoMcA committed:
fix(community): don't try to access localstorage on the server

LeoMcA committed:
fix(community): avoid hydration errors with localstorage cache

tried fixing this within the cache itself, but couldn't manage it

LeoMcA committed:
fix(community): randomly select contributors to display

custom elements are lit!
also workaround issue in contributor circle with links not being clickable

https://mozilla-hub.atlassian.net/browse/MP-1480

LeoMcA committed:
Merge remote-tracking branch 'origin/main' into MP-1369-new-community-page

LeoMcA committed:
fix(community): stylelint

LeoMcA committed:
Merge remote-tracking branch 'origin/main' into MP-1369-new-community-page

LeoMcA committed:
fix(community): reduce width

LeoMcA committed:
wip(community): various fixes

LeoMcA committed:
Merge remote-tracking branch 'origin/main' into MP-1369-new-community-page

LeoMcA committed:
wip(community): fix tsc

LeoMcA committed:
wip(community): fix line clamp

fiji-flo committed:
Merge remote-tracking branch 'upstream/main' into MP-1369-new-community-page

fiji-flo committed:
fix(community): use md to build

fiji-flo committed:
fix(community): remove hack

fiji-flo committed:
fix(community): use hyData format

LeoMcA committed:
Merge remote-tracking branch 'origin/main' into MP-1369-new-community-page

LeoMcA committed:
fix(community): offset-path url not working in chrome

LeoMcA committed:
fix(community): update copy

LeoMcA committed:
fix(community): keep things within main box

LeoMcA committed:
fix(community): run svgs through svgo

LeoMcA committed:
fix(community): stylelint

LeoMcA committed:
fix(community): avoid '| Contribute to MDN' title

fiji-flo commented on libs/constants/index.js:
```diff
@@ -149,6 +153,7 @@ export const CSP_DIRECTIVES = {
     "firefoxusercontent.com",
     "profile.stage.mozaws.net",
     "profile.accounts.firefox.com",
+    "https://github.com",
```
That's not needed anymore, is it?

caugner commented on client/src/community/contributor-list.ts:
```diff
@@ -0,0 +1,84 @@
+
+function popRandom<T>(array: Array<T>) {
+  const i = Math.floor(Math.random() * array.length);
+  return array.splice(i, 1)[0];
```
```suggestion
  return array.at(i);
```

LeoMcA replied:
Because I need to mutate the array, otherwise I could fetch the same contributor twice. It's `popRandom` not `atRandom`

LeoMcA committed:
Merge remote-tracking branch 'origin/main' into MP-1369-new-community-page

LeoMcA committed:
fix(community): fix tabbing order in contributor circle

LeoMcA committed:
wip(community): fix stylelint

LeoMcA committed:
wip(community): clip-path: inset(); doesn't work properly on webkit

LeoMcA committed:
wip(community): review updates

LeoMcA committed:
Merge remote-tracking branch 'origin/main' into MP-1369-new-community-page

LeoMcA committed:
fix(community): show focus ring on youtube link

LeoMcA committed:
fix(community): fix lint

LeoMcA committed:
fix(community): fix for https://github.com/mdn/yari/pull/11930

LeoMcA committed:
Merge remote-tracking branch 'origin/main' into MP-1369-new-community-page

LeoMcA committed:
fix(community): further fix for https://github.com/mdn/yari/pull/11930
